main.c

```diff
@@ -68,49 +67,31 @@ int main (int argc, char **argv) {
   }
 
   if (pulse) {
+    StartPulse(r, g, b, start_led, end_led);
 
-    sa.sa_handler = Handler;
-    sigemptyset(&sa.sa_mask);
-
-    if (sigaction(SIGALRM, &sa, NULL) == -1) {
-      perror("sigaction failure");
-      exit(1);
-    }
-
-    timer.it_value.tv_sec = 0;
-    timer.it_value.tv_usec = 10000;
-    timer.it_interval = timer.it_value;
-
-    if (setitimer(ITIMER_REAL, &timer, NULL) == -1) {
-      perror("error calling setitimer()");
-      exit(1);
+    uint8_t frame[kNumLeds * 4];
+    while (true) {
+      FrameUpdate(frame);
+      RenderFrame(frame);
+      usleep(10000);
```

Comment:
My problem with this is that the synchronization can shift over time, as usleep won't sleep for exactly the time we ask it to. So if usleep keeps sleeping for a little extra each time, we won't have the number of pulses we're expecting.

I'm not sure if the situation is better with the interval stuff, I guess we'd have to do a test.

Reply:
It's true, specially if the system is under significant load. To the best of my knowledge, a better solution might be to measure the loop time with a high-res clock and sleep tiny amounts until the 10ms go by. I'd also drop [that line where we are setting scheduling priority](https://github.com/lamassu/lamassu-led/blob/9fc7fe86e37cfe73c0489c5fd1b45ae739182e16/main.c#L47). Did you had good reason to use it before? I think it affects the global scheduler. Considering the suggestion I'm going above, it might have undesired consequences for our main machine process.